app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt

```diff
@@ -35,50 +33,51 @@ class ProfileFragment : Fragment() {
                               savedInstanceState: Bundle?): View? {
         rootView = inflater.inflate(R.layout.fragment_profile, container, false)
 
-        if (!authService.isLoggedIn())
+        if (!profileFragmentViewModel.isLoggedIn())
             redirectToLogin()
 
         rootView.logout.setOnClickListener {
-            authService.logout()
+            profileFragmentViewModel.logout()
             redirectToMain()
         }
 
+        profileFragmentViewModel.progress.observe(this, Observer {
+            it?.let { rootView.progressBar.isIndeterminate = it }
+        })
+
+        profileFragmentViewModel.visibility.observe(this, Observer {
+            it?.let { rootView.progressBar.visibility = it }
+        })
```

Comment:
Visibility and determinate should be combined. ViewModel should not know about the semantics of view

Reply:
How should I combine these 2 mutableLiveData ?

Reply:
No need to combine LiveData, see how it is handled in Login and Event fragment